tests/test_twitter_ads_bookmark.py

```diff
@@ -1,185 +1,185 @@
-                        "INVALID EXPECTATIONS\t\tSTREAM: {} REPLICATION_METHOD: {}".format(
-                            stream, expected_replication_method)
-                    )
-
-                # Verify at least 1 record was replicated in the second sync
-                self.assertGreater(
-                    second_sync_count, 0, msg="We are not fully testing bookmarking for {}".format(stream))
+import tap_tester.connections as connections
+import tap_tester.runner as runner
+import tap_tester.menagerie as menagerie
+
+from base import TwitterAds
+
+class BookmarkTest(TwitterAds):
+    """Test tap sets a bookmark and respects it for the next sync of a stream"""
+
+    def name(self):
+        return "tap_tester_twitter_ads_bookmark_test"
+
+    def test_run(self):
+        """
+        Verify that for each stream you can do a sync which records bookmarks.
+        That the bookmark is the maximum value sent to the target for the replication key.
+        That a second sync respects the bookmark
+            All data of the second sync is >= the bookmark from the first sync
+            The number of records in the 2nd sync is less then the first (This assumes that
+                new data added to the stream is done at a rate slow enough that you haven't
+                doubled the amount of data from the start date to the first sync between
+                the first sync and second sync run in this test)
+        Verify that for full table stream, all data replicated in sync 1 is replicated again in sync 2.
+        PREREQUISITE
+        For EACH stream that is incrementally replicated there are multiple rows of data with
+            different values for the replication key
+        """
+
+        streams_to_test = self.expected_streams()
+
+        # For following streams(except targeting_tv_markets and targeting_tv_shows), we are not able to generate any records.
+        # targeting_tv_markets and targeting_tv_shows streams take more than 5 hour to complete the sync.
+        #  So, skipping those streams from test case.
+        streams_to_test = streams_to_test - {'cards_image_conversation', 'cards_video_conversation', 'cards_image_direct_message',
+                                            'cards_video_direct_message', 'accounts_daily_report', 'campaigns_daily_report', 'accounts',
+                                            'targeting_tv_markets', 'targeting_tv_shows'}
+
+        # Endpoints are swapped for content_categories and iab_categories streams - https://jira.talendforge.org/browse/TDL-18374        
+        streams_to_test = streams_to_test - {'iab_categories', 'content_categories'}
+
+        # Invalid endpoint for targeting_events stream - https://jira.talendforge.org/browse/TDL-18463
+        streams_to_test = streams_to_test - {'targeting_events'}
+
+        # Invalid bookmark for tweets stream - https://jira.talendforge.org/browse/TDL-18465
+        streams_to_test = streams_to_test - {'tweets'}
```

Comment:
Please make sure we remove these things in respective PRs or once all PR gets merged.

Reply:
Yes, we will remove these work arounds once all PR gets merged.